packages/accordion/src/vaadin-accordion-heading.d.ts

```diff
@@ -0,0 +1,58 @@
+/**
+ * @license
+ * Copyright (c) 2019 - 2022 Vaadin Ltd.
+ * This program is available under Apache License Version 2.0, available at https://vaadin.com/license/
+ */
+import { ActiveMixin } from '@vaadin/component-base/src/active-mixin.js';
+import { DirMixin } from '@vaadin/component-base/src/dir-mixin.js';
+import { ThemableMixin } from '@vaadin/vaadin-themable-mixin/vaadin-themable-mixin.js';
+
+/**
+ * The accordion heading element.
+ *
+ * `vaadin-accordion-heading` is the element for the headings in the accordion.
+ * As recommended by the WAI ARIA Best Practices, each heading needs to wrap a
+ * `<button>`. This element puts that button in the Shadow DOM, as it is more
+ * convenient to use and doesn’t make styling of the heading more problematic.
+ *
+ * The WAI ARIA Best Practices also recommend setting `aria-level` depending
+ * on what level the headings are. It is hard to determine the level of a heading
+ * algorithmically, and setting it is not strictly required to have an accessible
+ * accordion. To keep things easier to use, this element does not set `aria-level`
```

Comment:
Are you planning to add an API for it on the flow side?

Reply:
Yes, that's the idea. For each `AccordionPanel`, we would create an instance of `AccordionHeading` on the server side and then expose it e.g. `getHeadingComponent()` API, to enable setting `aria-level` e.g. like this:

```java
Accordion accordion = new Accordion();
AccordionPanel panel = accordion.add("Heading text", new Div());
panel.getHeadingComponent().getElement().setAttribute("aria-level", "3");
```

Does that sound reasonable? Or do you have any other suggestions on how the API might look like?

Reply:
Thanks for already considering it! This would reduce the boilerplate I've created in my apps where I've manually added h* elements inside the header. 

I'm totally fine with the API, but I'm wondering if people would find / use them. Would it make sense to have something like an AriaLevel enum with 1-6 allowing to create constructors like: 

```
new Details(String caption, Component content) 

new Details(String caption, AriaLevel headingLevel, Component content) 

new Details(String caption, AriaLevel headingLevel, Component content) 

// AriaLevel could be optional / nullable allowing it to be not set if not present like in the first constructor
```

Reply:
Totally valid! Just checked my application and I have only accordions in use as of now, so I'm totally fine with your idea 👍